### PR TITLE
Sl 37

### PR DESCRIPTION
**Pull request by grant-freeman**
This PR includes commits from SL-34 and SL-37 because I needed dev changes from SL-34 to finish my story. I've finished writing a startup script for our api, a massive function for our update, modifications to our detection model, and tests. Tests must be made with the api running.

### CODE REVIEW

**Review by slimwizard — CHANGES_REQUESTED** [review]
99% of it looks good. All I ask is that the body of the receive_image API method be broken up into separate functions like 'split_image', 'enhance_image', 'process_image'. It just doesn't make much sense from both a manageability and a readability standpoint to have a gigantic block of code handle all of those different tasks like that. Also, not sure if this was you or Charley but the Spots table's 'occupied' column and all references to it was changed to 'availability'. I don't think a parking spot's availability can be logically represented as true or false. It works better as 'occupied' or 'is_occupied', or even just 'available'.

`if (spot.occupied): do this` makes much more sense than `if (spot.availibility): do this`

Everything else looks good.

File: image-processing-server/detection.py [hunk 1]
```diff
@@ -102,6 +102,7 @@
 for (x,y,w,h) in cars:
 	cv2.rectangle(img,(x,y),(x+w-20,y+h-20),(0,0,255),2)      
 
-cv2.imshow('image', img)
-cv2.waitKey(300000)
-
+if cars.size:
```
Comment (Jfrachal) [inline image-processing-server/detection.py:105]: Can you clarify what this method is doing? I'm unsure what's going on here in the context of the code. 

cars.size

**Review by Jfrachal — APPROVED** [review]
Good code, only issue is getting some clarifying comments where noted. No significant issues, good to merge.

File: image-processing-server/detection.py (hunk 1 above)
Comment (grant-freeman) [reply image-processing-server/detection.py:105]: if the size of the cars array is not 0, basically

**Comment by slimwizard** [comment]
Looks good

**Pull request merged**